### PR TITLE
Adding statistics for admins

### PR DESCRIPTION
**Pull request by JimVonLarsson**

### CODE REVIEW

File: src/AppBundle/Controller/Admin/StatisticsController.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        "11" => "November",
+        "12" => "December"
+    ];
+
+    /**
+     * @Route("/", name="admin_statistics")
+     */
+    public function indexAction(Request $request)
+    {
+        $city = $request->query->get("city", 2);
```
Comment (jongotlin) [inline src/AppBundle/Controller/Admin/StatisticsController.php]: Please don't hard code value 2. Why id 2? Can you go for `$cityRepository->findAll()[0]`?

File: src/AppBundle/Controller/Admin/StatisticsController.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        foreach ($connections as $connection) {
+            $year = date("Y", strtotime($connection["created_at"]));
+            if (!in_array($year, $years)) {
+                $years[] = $year;
+            }
+        }
+        return $years;
+    }
+
+    protected function structuredMatches($matches)
```
Comment (jongotlin) [inline src/AppBundle/Controller/Admin/StatisticsController.php]: Do you think you can put allt this in the query using count(*) and group by?

File: src/AppBundle/Entity/ConnectionRepository.php [hunk 3]
```diff
@@ -6,6 +6,50 @@
...
+    /**
+     * @param String $city
+     * @param String $year
+     * @param String $type
+     * @return array
+     */
+    public function getMatches($city, $year, $type)
+    {
+        // set up query for city_id
+        $cityQuery = (isset($city)) ? "AND c.city_id = trim({$city})" : "";
```
Comment (jongotlin) [inline src/AppBundle/Entity/ConnectionRepository.php]: Isn't `$city` always set? `!empty` maybe?

**Comment by jongotlin** [comment]
Thanks! Just some minor comments

**Comment by JimVonLarsson** [comment]
Thanks @jongotlin! I´ll take a look later today :)

File: src/AppBundle/Controller/Admin/StatisticsController.php (hunk 2 above)
Comment (JimVonLarsson) [inline src/AppBundle/Controller/Admin/StatisticsController.php]: I'm not sure I know a better solution for this.. Do you got any best practice on how to populate the year selection?

Comment (jongotlin) [inline src/AppBundle/Controller/Admin/StatisticsController.php]: I believe you can do something like this
```php
->select('COUNT(*) as cnt, SUBSTRING(c.createdAt, 1, 4) as year')
->groupBy('year')
```

Comment (jongotlin) [inline src/AppBundle/Controller/Admin/StatisticsController.php]: Using dql that is. Not sql.

Comment (mikaelohren) [inline src/AppBundle/Controller/Admin/StatisticsController.php]: Can we merge or is this important to fix?

Comment (jongotlin) [inline src/AppBundle/Controller/Admin/StatisticsController.php]: I'd say it's a blocker. Also it should be quite easy to fix.

Comment (JimVonLarsson) [inline src/AppBundle/Controller/Admin/StatisticsController.php]: I'll take a look later today :)

Comment (JimVonLarsson) [inline src/AppBundle/Controller/Admin/StatisticsController.php]: @jongotlin I'm no big fan of magic, so i skipped dql and went for sql..

**Comment by jongotlin** [comment]
Dql is no magic :-) But since you're only fetching a date it's ok to use sql. But you have to use prepared statement to prevent sql injections.

**Comment by mikaelohren** [comment]
So using a prepared statement is the last fix before merge?

**Comment by jongotlin** [comment]
Yes. Can you do it @JimVonLarsson?

**Comment by JimVonLarsson** [comment]
Sure!

**Comment by JimVonLarsson** [comment]
Have you checked the changes yet @jongotlin?

**Comment by jongotlin** [comment]
Thanks! Maintainer doesn't get notified for every commit so I haven't seen you updated it.

**Pull request merged**